dotnet/src/Extensions/Planning.ActionPlanner/skprompt.txt

```diff
@@ -0,0 +1,12 @@
+A planner takes a list of functions, a goal, and chooses which function to use.
+For each function the list includes details about the input parameters.
+{{basicPlanner.GoodExamples}}
+{{basicPlanner.EdgeCaseExamples}}
+[END OF EXAMPLES]
+[REAL SCENARIO STARTS HERE]
+- List of functions:
+{{basicPlanner.ListOfFunctions}}
+- End list of functions.
+Goal: {{ $input }}
+{"plan":{
+"rationale": 
```

Comment:
Do we want it to generate rationale every time, or could this be a "debug mode" optional feature? Just thinking that, if this really is the most basic, cheap planner, minimizing token consumption is important as it could be called many, many times, including in tight loops.

Reply:
this is a request we've had from users, the planner explain why it's making some choices. It's also a common feature in other frameworks